CMakeLists.txt

```diff
@@ -0,0 +1,356 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(coap VERSION 4.3.0 LANGUAGES CXX C)
+
+set(LIBCOAP_API_VERSION 2)
+
+#
+# options to tweak the library
+#
+
+option(ENABLE_DTLS "Enable building with DTLS support" ON)
```

Comment:
With ENABLE_DTLS=ON and with WITH_OPENSSL. WITH_GNUTLS and WITH_TINYDTLS all set to OFF no encryption support is enabled.  configure.ac checks for GnuTLS and then OpenSSL in this case.